### PR TITLE
add bluepill mcu to splittest

### PR DESCRIPTION
**Pull request by dvermd**
This adds the bluepill MCU to the splittest keyboard 

## Description

The MCU is added, some keymaps are created for bitbang, serial halfduplex and serial fullduplex communication between halves.

The move of `RGBLIGHT_ENABLE` for teensy2 and promicro has not been tested as I don't own these MCU however the diff seems simple enough to be confident in the change.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/splittest/bluepill/readme.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Wiring
+
+### Switches
+
+-   Add switches to both Bluepills across B10 and B1 pins
+
+### Handedness
+
+Have a look at the [handedness documentation](ttps://docs.qmk.fm/#/feature_split_keyboard?id=setting-handedness)
```
[the remark this answers is not in the document]
Comment (dvermd) [reply keyboards/handwired/splittest/bluepill/readme.md]: fixed in new commit

**Review by drashna — COMMENTED** [review]
the hardware config files should be handled at the kb level, not the keymap level.  Especially if stuff like the RP2040 is added here, since it doesn't use the STM32 config... 

And at the kb level, you could use post_config.h to change/set options based on usart driver or not.

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/splittest/teensy_2/rules.mk [hunk 2]
```diff
@@ -3,3 +3,5 @@ MCU = atmega32u4
 
 # Bootloader selection
 BOOTLOADER = halfkay
+
+RGBLIGHT_ENABLE = yes        # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/handwired/splittest/teensy_2/rules.mk:7]:
```suggestion
RGBLIGHT_ENABLE = yes
```

File: keyboards/handwired/splittest/rules.mk [hunk 3]
```diff
@@ -9,7 +9,7 @@ COMMAND_ENABLE = yes        # Commands for debug and configuration
 NKRO_ENABLE = no            # Enable N-Key Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 AUDIO_ENABLE = no           # Audio output
-RGBLIGHT_ENABLE = yes        # Enable WS2812 RGB underlight.
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/handwired/splittest/rules.mk:12]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

File: keyboards/handwired/splittest/promicro/rules.mk [hunk 4]
```diff
@@ -3,3 +3,5 @@ MCU = atmega32u4
 
 # Bootloader selection
 BOOTLOADER = caterina
+
+RGBLIGHT_ENABLE = yes        # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/handwired/splittest/promicro/rules.mk:7]:
```suggestion
RGBLIGHT_ENABLE = yes
```

File: keyboards/handwired/splittest/bluepill/rules.mk [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# MCU name
+MCU = STM32F103
+
+# Bootloader selection
+BOOTLOADER = stm32duino
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+CONSOLE_ENABLE = yes         # Console for debug
```
Comment (fauxpark) [inline keyboards/handwired/splittest/bluepill/rules.mk:9]:
```suggestion
CONSOLE_ENABLE = yes

# Enter lower-power sleep mode when on the ChibiOS idle thread
OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/splittest/bluepill/config.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2022 dvermd (@dvermd)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+#define BLUEPILL
+
```
Comment (fauxpark) [inline keyboards/handwired/splittest/bluepill/config.h:9]: I don't see where this is used.
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/splittest/bluepill/config.h (hunk 6 above)
Comment (dvermd) [reply keyboards/handwired/splittest/bluepill/config.h]: That's right, it's a leftover of the first version of this PR

**Review by keyboard-magpie — DISMISSED** [review]

**Comment by keyboard-magpie** [comment]
I'm actually having issues with compiling the serial_fullduplex and serial_halfduplex keymaps:

`avr-ar: .build/obj_handwired_splittest_promicro_serial_halfduplex/serial_usart.o: No such file or directory`

File: keyboards/handwired/splittest/bluepill/readme.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+-   Connect the B8 pins on both sides together
+
+#### serial - usart half duplex
+
+Check the [documentation](https://docs.qmk.fm/#/serial_driver?id=usart-half-duplex) to determine the pull-up resistor.
+
+-   To use the default usart (USART1) with the standard role pins,
+    -   Connect the A9 pins on both sides together and to add a pull-up resistor on one of these pins
+    -   Define USE_SDI1_ON_STANDARD_ROLE_PINS in config.h
+-   To use the default usart (USART1) with the alterante role pins,
```
Comment (keyboard-magpie) [inline keyboards/handwired/splittest/bluepill/readme.md:37]: "alternate" not alterante

File: keyboards/handwired/splittest/bluepill/readme.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
+# bluepill bitbang splittest
```
Comment (keyboard-magpie) [inline keyboards/handwired/splittest/bluepill/readme.md:1]: I think in general this section could do with a starter paragraph "Make example for this keyboard (after setting up your build environment):"

**Review by keyboard-magpie — CHANGES_REQUESTED** [review]
It took a bit of poking and prodding to get this to compile without chibios warnings- some of that may well be related to setting defines at the wrong level- but clarifying at what level to set the ROLE_PINS or USE_SDI2 would certainly make it clearer for end users.

File: keyboards/handwired/splittest/bluepill/readme.md (hunk 7 above)
Comment (dvermd) [reply keyboards/handwired/splittest/bluepill/readme.md:37]: fixed

File: keyboards/handwired/splittest/bluepill/readme.md (hunk 8 above)
Comment (dvermd) [reply keyboards/handwired/splittest/bluepill/readme.md]: Does the updated documentation explains the parts you were missing ?

**Review by keyboard-magpie — APPROVED** [review]
Looks good, thanks for improving the readme

**Pull request merged**